executing/executing.py

```diff
@@ -309,6 +310,142 @@ def executing(cls, frame_or_tb):
             lineno = frame.f_lineno
             lasti = frame.f_lasti
 
+        def check_code(code):
+
+            positions = code.co_positions()
+            bc = list(dis.Bytecode(code, show_caches=True))
+
+            for pos, inst in zip(positions, bc):
+                if inst.starts_line is not None:
+                    assert pos[0] <= inst.starts_line <= pos[1]
+
+        #        check_code(frame.f_code)
+
+        if sys.version_info >= (3, 11):
+            # we can use co_positions() since 3.11, which has fewer limitations
+
+            positions = list(frame.f_code.co_positions())
+            source = cls.for_frame(frame)
+            stmts = source.statements_at_line(lineno)
+
+            def find_node(index, lineno=True):
+                position = positions[index // 2]
+                return find_node_by_position(position, lineno=lineno)
+
+            def find_node_by_position(position, lineno=True):
+                if lineno:
+
+                    def match(a, b):
+                        return a == b
+
+                else:
+
+                    def match(a, b):
+                        return a[1:] == b[1:]
+
+                result = only(node
+                              for node in source._nodes_by_line[position[0]]
+                              if isinstance(node, (ast.expr, ast.stmt))
+                              if not isinstance(node, ast.Expr) if match(
+                                  position,
+                                  (
+                                      node.lineno,
+                                      node.end_lineno,
+                                      node.col_offset,
+                                      node.end_col_offset,
+                                  ),
+                              ))
+
+                return result
+
+            bc_list = list(dis.Bytecode(frame.f_code, show_caches=True))
+
+            try:
+                node = find_node(lasti)
+
+            except:
+                # lineno of LOAD_METHOD instructions get set to end_lineno by the python compiler
+                # propably to achive better error messages
+                # this is a work around 
+                if bc_list[lasti // 2].opname == "LOAD_METHOD":
```

Comment:
This also applies to LOAD_ATTR and STORE_ATTR.

Also from this point onwards I can't understand what the code is doing, please add some comments. That will make it easier for me to help you.

Reply:
I'm currently in the process to understanding it myself. It might be that there are still missing cases.  I will let you know what the code does an write comments.